syntax = "proto3";

option go_package = "github.com/wholesome-ghoul/persona-prototype-6/protos";

service Content {
  rpc Personalize(Contents) returns (ContentScheme) {}
  rpc InsertContents(Contents) returns (Status) {}
  rpc Rollback(ContentIDs) returns (Status) {}
  rpc SemanticSearch(Search) returns (ContentIDs) {}
  rpc CalculateSpamProbs(Sentences) returns (SpamProbs) {}
}

message Empty { }

message ContentScheme {
  string content_id = 1;
  float spam = 2;
  map<string, float> emotions = 3;
  map<string, float> personalities = 4;
}

message Contents {
  repeated string ids = 1;
  repeated string texts = 2;
}

message ContentIDs {
  repeated string items = 1;
}

message Search { 
  string text = 1;
}

message Sentences { 
  repeated string items = 1;
}

message SpamProbs { 
  repeated float items = 1;
}

message Status {
  int32 code = 1;
}
